dataprofiler/tests/profilers/profiler_options/test_histogram_and_quantiles_option.py

```diff
@@ -173,18 +259,46 @@ def test_eq(self):
         self.assertNotEqual(options, options2)
         options2.bin_count_or_method = "sturges"
         self.assertEqual(options, options2)
+        options.num_quantiles = 30
+        self.assertNotEqual(options, options2)
+        options2.num_quantiles = 50
+        self.assertNotEqual(options, options2)
+        options2.num_quantiles = 30
+        self.assertEqual(options, options2)
 
     def test_json_encode(self):
-        option = HistogramOption(is_enabled=False, bin_count_or_method="doane")
+        option = HistogramAndQuantilesOption(
+            is_enabled=False, bin_count_or_method="doane"
+        )
 
         serialized = json.dumps(option, cls=ProfileEncoder)
 
         expected = {
-            "class": "HistogramOption",
+            "class": "HistogramAndQuantilesOption",
             "data": {
                 "bin_count_or_method": "doane",
+                "num_quantiles": 1000,
                 "is_enabled": False,
             },
         }
 
         self.assertDictEqual(expected, json.loads(serialized))
+
+    def test_json_decode_warn(self):
+        expected = {
+            "class": "HistogramOption",
+            "data": {
+                "bin_count_or_method": "doane",
+                "is_enabled": False,
+            },
+        }
+
+        expected_string = json.dumps(expected)
+
+        expected_warning = (
+            "HistogramOption will be deprecated in the future."
+            "Please begin utilizing the new HistogramAndQuantilesOption class."
+        )
+
+        with self.assertWarnsRegex(DeprecationWarning, expected_warning):
+            load_option(json.loads(expected_string))
```

Comment:
anything else we need to validate to ensure this is functional

Reply:
should it do that, though.... 🤔